src/Tree/Entity/Repository/NestedTreeRepository.php

```diff
@@ -140,7 +140,16 @@ public function getRootNodesQueryBuilder($sortByField = null, $direction = 'asc'
         ;
 
         if (null !== $sortByField) {
-            $qb->orderBy('node.'.$sortByField, 'asc' === strtolower($direction) ? 'asc' : 'desc');
+            if (is_array($sortByField)) {
+                foreach ($sortByField as $key => $field) {
+                    $fieldDirection = is_array($direction) ? ($direction[$key] ?? 'asc') : $direction;
+                    if (($meta->hasField($field) || $meta->isSingleValuedAssociation($field)) && in_array(strtolower($fieldDirection), ['asc', 'desc'], true)) {
```

Comment:
Let's move the cheaper check first:
```suggestion
                    if (in_array(strtolower($fieldDirection), ['asc', 'desc'], true) && ($meta->hasField($field) || $meta->isSingleValuedAssociation($field))) {
```
BTW, you are introducing a validation about the allowed sort directions here and omitting the sort if the passed value is not in the expected values. I suggest to update this to mimic the previous behavior, where if the passed value is not in the allowed list, a default one is used.